src/webgl/shaders/lighting.glsl

```diff
@@ -67,6 +79,96 @@ LightResult _light(vec3 viewDirection, vec3 normal, vec3 lightVector) {
   return lr;
 }
 
+// converts the range of "value" from [min1 to max1] to [min2 to max2]
+float map(float value, float min1, float max1, float min2, float max2) {
+  return min2 + (value - min1) * (max2 - min2) / (max1 - min1);
+}
+
+/*
+vec2 mapTextureToNormal( vec3 normal ){
+  // x = r sin(phi) cos(theta)
+  // y = r cos(phi)
+  // z = r sin(phi) sin(theta)
+  float phi = acos( normal.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  if(  abs(sin(phi)) > 0.0 ){
+    theta = acos( normal.x / sin(phi) );
+  }   
+
+  vec2 newTexCoor = vec2(
+    map(theta, 0., PI, 0., 1.),
+    map(phi, 0., PI, 1., 0.)
+  );
+
+  return newTexCoor;
+}
+*/
+
+vec2 nToE( vec3 v ){
+  // x = r sin(phi) cos(theta)   
+  // y = r cos(phi)  
+  // z = r sin(phi) sin(theta)
+  float phi = acos( v.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  theta = acos(v.x / sin(phi));
+  float sinTheta = v.z / sin(phi);
+  if (sinTheta < 0.0) {
+    // Turn it into -theta, but in the 0-2PI range
+    theta = 2.0 * PI - theta;
+  }
+  theta = theta / (2.0 * 3.14159);
+  phi = phi / 3.14159 ;
+  
+  vec2 angles = vec2( phi, theta );
+  return angles;
+}
+
+
+vec3 calculateImageDiffuse( vec3 vNormal, vec3 vViewPosition ){
+  // make 2 seperate builds 
+  vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);  // hardcoded world camera position
+  vec3 worldNormal = normalize(vNormal);
+  vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
+  vec3 R = reflect(lightDirection, worldNormal);
+  // use worldNormal instead of R
+  // added nToE instead of this, remove this completely if the
+  // new function works
+  // vec2 newTexCoor = mapTextureToNormal( R );
+  vec2 newTexCoor = nToE( R );
+  vec4 texture = TEXTURE( environmentMapDiffused, newTexCoor );
+  // return texture.xyz;
+  // this is to make the darker sections more dark
+  // png and jpg usually flatten the brightness so it is to reverse that
+  return pow(texture.xyz, vec3(10.0));
```

Comment:
I tried changing `imageLight()` with `lights()` in a sketch to see if they could be roughly equivalent. I think 10 might be a bit too aggressive for the diffuse light unfortunately. I tried playing around with some other curves:

<table>
<tr>
<th>Lighting call</th>
<th>

```glsl
pow(
  texture.xyz,
  vec3(10.0)
);
```

</th>
<th>

```glsl
texture.xyz;
```

</th>
<th>

```glsl
smoothstep(
  vec3(0.),
  vec3(0.8),
  texture.xyz
);
```

</th>
<tr>
<td>

```js
imageLight(img);
```

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/fb2a5f09-d4d0-4ff7-90a2-0b632311cc32)

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/2f8a27a2-3c38-4965-b0b2-d73a16934c0f)

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/bb1b9140-fe8a-466e-be57-dba171ac8e78)

</td>
</tr>
<tr>
<td>

```js
lights()
```

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/7cf85b78-705d-45fc-bb2b-d178d546e55f)

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/7cf85b78-705d-45fc-bb2b-d178d546e55f)

</td>
<td>

![image](https://github.com/processing/p5.js/assets/5315059/7cf85b78-705d-45fc-bb2b-d178d546e55f)

</td>
</tr>
</table>

So although it's not physically accurate or anything, maybe doing this will help make it easier for people to switch from `lights()` to `imageLight()`? This maps texture values from 0-0.8 to 0-1 along a smooth curve:
```suggestion
  return smoothstep(vec3(0.0), vec3(0.8), texture.xyz);
```

Reply:
Done for `caluculateImageDiffuse`, should I do this for `calculateImageSpecular` are well?

Reply:
I think specular looks ok how it is, the inconsistency is fine as long as it looks ok.